client/my-sites/plans/plan-overview/plan-remove/index.jsx

```diff
@@ -8,54 +8,48 @@ import React from 'react';
 /**
  * Internal dependencies
  */
+import { cancelSitePlanTrial } from 'state/sites/plans/actions';
 import CompactCard from 'components/card/compact';
+import { connect } from 'react-redux';
 import Dialog from 'components/dialog';
+import { getPlansBySite } from 'state/sites/plans/selectors';
 import { isInGracePeriod } from 'lib/plans';
 import notices from 'notices';
 import paths from '../../paths';
-import { fetchSitePlansCompleted } from 'state/sites/plans/actions';
-import wpcom from 'lib/wp';
 
 const PlanRemove = React.createClass( {
 	propTypes: {
+		cancelSitePlanTrial: React.PropTypes.func.isRequired,
+		isUpdating: React.PropTypes.bool.isRequired,
 		plan: React.PropTypes.object.isRequired,
 		selectedSite: React.PropTypes.oneOfType( [
 			React.PropTypes.object,
 			React.PropTypes.bool
-		] ).isRequired,
-		store: React.PropTypes.object.isRequired
+		] ).isRequired
 	},
 
 	getInitialState() {
 		return {
-			isCanceling: false,
 			showDialog: false
 		};
 	},
 
 	removePlan( closeDialog ) {
-		this.setState( { isCanceling: true } );
+		this.props.cancelSitePlanTrial( this.props.selectedSite.ID, this.props.plan.id ).then( () => {
+			Dispatcher.handleViewAction( {
```

Comment:
I think this could be called as a props function defined in `connect`. Then you would just call `this.props.fetchSites()` here. The advantage is grouping all the methods accessing the store together and better understand how `plans.isUpdating` is being updated.